vendingMachine.js

```diff
@@ -1,68 +1,95 @@
+// E-codes explained in README.MD file
 const prompt = require('prompt-sync')();
-let products = [
+const productsMachine = [
   ["Chocorramo", "Gansito", "Galletas"],
   ["Maizitos", "Tostacos", "Yogurt"],
   ["Papas", "Platanitos", "Ponque"]
 ];
-let stock = [
-  [5, 5, 5],
-  [5, 5, 5],
-  [5, 5, 5]
+const productStock = [
+  ["20", "5", "20"],
+  ["5", "1000", "5"],
+  ["100", "5", "5"]
 ];
 
-function showVendingMachine(arr1, arr2) {
-  let result = '';
-  let coord, stoc;
-  for (let i = 0; i < arr1.length; i++) { //0
-    for (let j = 0; j < arr1[i].length; j++) { //2
-      result = result + arr1[i][j] + '    ';
-      if (j === arr1.length - 1) {
-        coord = ('   ' + (i + 1) + '.' + (j - 1) + '       ' + (i + 1) + '.' + (j) + '       ' + (i + 1) + '.' + (j + 1))
-          //coord = ((i + 1) + '.' + (j - 1) + (i + 1) + '.' + (j) + (i + 1) + '.' + (j + 1))
-        stoc = ('    ' + arr2[i][j - 2] + '         ' + arr2[i][j - 1] + '         ' + arr2[i][j]);
-        //stoc = (arr2[i][j] + arr2[i][j + 1] + arr2[i][j + 2]);
-        result = result + `\n${coord}\n${stoc}\n`;
+function showVendingMachine(matrixProducts, matrixStock, matrixCode) {
+  const SPACES = " ";
+  const supline = "-";
+  const topLine = supline.repeat(122);
+  const spaceTable = 40; //E1 
+  const codeProduct = 3; //E2 
+  let itemTable = '';
+  let coord, stock, spaceLeft, spaceLeft1, spaceLeft2, spaceLeft3;
+  let spaceRight, spaceRight1, spaceRight2, spaceRight3;
+  console.log(topLine);
+  for (let i = 0; i < matrixProducts.length; i++) {
+    for (let j = 0; j < matrixProducts[i].length; j++) {
+      let imparLengthItem = 0;
+      let leftSpaceTable, rightSpaceTable, squareItemTable;
+      leftSpaceTable = SPACES.repeat(Math.round((spaceTable - (matrixProducts[i][j].length)) / 2));
+      if (matrixProducts[i][j].length % 2 === 1) {
+        imparLengthItem = 1;
+      }
+      rightSpaceTable = SPACES.repeat(Math.round((spaceTable - (matrixProducts[i][j].length)) / 2) - imparLengthItem);
+      squareItemTable = `|${leftSpaceTable}${matrixProducts[i][j]}${rightSpaceTable}`;
+      if (matrixStock[i][j] <= 1) {
+        matrixProducts[i][j] = `${supline.repeat(4)}`;
+      }
+      itemTable = itemTable + `${squareItemTable}`;
+      if (j === matrixProducts.length - 1) {

```

Comment:
Please add a comment explaining why this if. 
For example
```suggestion
// All the visual styles added when the final of the row is identified
      if (j === matrixProducts.length - 1) {

```

Reply:
Done